cmd/ro/main.go

```diff
@@ -75,9 +82,14 @@ func registerFlags() {
 	flag.StringVar(&userEnv, "userenv", "RO_USER", "env variable for user name")
 	flag.StringVar(&pswdEnv, "pswdenv", "RO_PASS", "env variable for user password")
 	flag.DurationVar(&pollInterval, "poll-interval", time.Second, "interval for polling an outstanding order (set 0 to disable polling)")
+	flag.StringVar(&pubKeyPath, "pubkey", "id_rsa.pub", "path to SSH public key")
 }
 
 func readLine(prompt string) (line string, err error) {
+	// TODO: find a way to ask for username and password
+	// such that the prompt itself is not captured in the
+	// standard output. This way running `ro ... ssh-agent`
+	// will set the SSH_AUTH_PATH environment variable.
 	fmt.Printf(prompt)
```

Comment:
`fmt.Fprintf(os.Stderr, prompt)` would write to stderr, not stdout.

Reply:
Thanks! Fixed it.
Though unfortunately, since ro-ssh-agent doesn't daemonize, the "export ..." command doesn't get executed by bash until the process exits. Do you know what's the best way to daemonize Go programs?

Reply:
This probably won't work the way you think it does. A process's environment is a copy of the parent process's environment, with changes only occurring in that copy; the changes won't be propagated to bash. With ssh-agent, you have to call `eval $(ssh-agent)`; with [`keychain(1)`](https://www.funtoo.org/Keychain), you have to source a file where the environment variables have been written (e.g. `keychain -q && source $HOME/.keychain/$(host -s)-sh`).

Reply:
Exactly, I wanted to mimic ssh-agent's behavior by doing `eval $(ro ... ssh-agent)$`.

Perhaps something like [this](https://wiki.archlinux.org/index.php/SSH_keys#Start_ssh-agent_with_systemd_user) would be better though, since then it works everywhere.